src/scanner/images/index.ts

```diff
@@ -13,30 +13,43 @@ import {
   LegacyPluginResponse,
 } from './docker-plugin-shim';
 
-export async function pullImages(images: IPullableImage[]): Promise<IPullableImage[]> {
-  const pulledImages: IPullableImage[] = [];
-
-  for (const image of images) {
-    const { imageName, imageWithDigest, fileSystemPath } = image;
-    if (!fileSystemPath) {
-      continue;
-    }
-
+/*
+ pulled images by skopeo archive repo type:
+ 1st try to pull by docker archive image if it fail try to pull by oci archive
+*/
+async function pullImageBySkopeoRepo(imageToPull: IPullableImage): Promise<IPullableImage> {
+    // Scan image by digest if exists, other way fallback tag
+    const scanId = imageToPull.imageWithDigest ?? imageToPull.imageName;
+    imageToPull.skopeoRepoType = SkopeoRepositoryType.DockerArchive;
     try {
-      // Scan image by digest if exists, other way fallback tag
-      const scanId = imageWithDigest ?? imageName;
-      await skopeoCopy(scanId, fileSystemPath);
-      pulledImages.push(image);
-    } catch (error) {
-      logger.error({error, image: imageWithDigest}, 'failed to pull image');
+        // copy docker archive image
+        await skopeoCopy(scanId, imageToPull.fileSystemPath, imageToPull.skopeoRepoType);
+    } catch (dockerError) {
+        imageToPull.skopeoRepoType = SkopeoRepositoryType.OciArchive;
+        // copy oci archive image
+        await skopeoCopy(scanId, imageToPull.fileSystemPath, imageToPull.skopeoRepoType);
     }
-  }
+    return imageToPull;
+}
 
-  return pulledImages;
+export async function pullImages(images: IPullableImage[]): Promise<IPullableImage[]> {
+    const pulledImages: IPullableImage[] = [];
+    for (const image of images) {
+        if (!image.fileSystemPath) {
+            continue;
+        }
+        try {
+            const pulledImage = await pullImageBySkopeoRepo(image);
+            pulledImages.push(pulledImage);
+        } catch (error) {
+            logger.error({error, image: image.imageWithDigest?? image.imageName}, 'failed to pull image docker/oci archive image');
+        }
+    }
+    return pulledImages;
 }
 
-export function getImagesWithFileSystemPath(images: IScanImage[]): IPullableImage[] {
-  return images.map((image) => ({ ...image, fileSystemPath: getDestinationForImage(image.imageName) }));
+export function getImagesWithFileSystemPath(images: IScanImage[]): { imageName: string; skopeoRepoType: SkopeoRepositoryType; fileSystemPath: string; imageWithDigest?: string }[] {
```

Comment:
I would suggest to create a type for this object
```js
{ imageName: string; skopeoRepoType: SkopeoRepositoryType; fileSystemPath: string; imageWithDigest?: string }
```
and use it here for readability purposes 🤗